promql/engine_test.go

```diff
@@ -1193,6 +1367,18 @@ func TestSubquerySelector(t *testing.T) {
 					},
 					Start: time.Unix(10, 0),
 				},
+				{
+					Query: "metric[20s::10s]",
+					Result: Result{
+						nil,
+						Matrix{Series{
+							Points: []Point{{V: 1, T: 0}, {V: 2, T: 10000}},
+							Metric: labels.FromStrings("__name__", "metric")},
+						},
+						nil,
+					},
+					Start: time.Unix(10, 0),
```

Comment:
As before, we need to cover cases where the evaluation-time alignment is actually different from the conventional alignment.